power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/y_bus.hpp

```diff
@@ -329,16 +340,96 @@ template <bool sym> class YBus {
         auto const& y_bus_entry_indptr = y_bus_struct_->y_bus_entry_indptr;
         // loop for each y bus position
         for (Idx entry = 0; entry != nnz(); ++entry) {
+            // start admittance accumulation with zero
+            ComplexTensor<sym> entry_admittance{0.0};
+            IdxVector entry_param_shunt;
+            IdxVector entry_param_branch;
+            // std::vector<std::pair<Idx, Idx>> entry_param_shunt_pair; //WIP
+            //   loop over all entries of this position
+            for (Idx element = y_bus_entry_indptr[entry]; element != y_bus_entry_indptr[entry + 1]; ++element) {
+                auto param_idx = y_bus_element[element].idx;
+                if (y_bus_element[element].element_type == YBusElementType::shunt) {
+                    // shunt
+                    entry_admittance += shunt_param[param_idx];
+                    entry_param_shunt.push_back(param_idx);
+                } else {
+                    // branch
+                    entry_admittance +=
+                        branch_param[param_idx].value[static_cast<Idx>(y_bus_element[element].element_type)];
+                    entry_param_branch.push_back(param_idx);
+                    // entry_param_shunt_pair.push_back(std::make_pair(param_idx, element)); //WIP
+                }
+            }
+            // assign
+            admittance[entry] = entry_admittance;
+            map_admittance_param_branch.push_back(entry_param_branch);
+            map_admittance_param_shunt.push_back(entry_param_shunt);
+            // map_admittance_param_shunt_pairs.push_back(entry_param_shunt_pair); //WIP
+        }
+        // move to shared ownership
+        admittance_ = std::make_shared<ComplexTensorVector<sym> const>(std::move(admittance));
+    }
+
+    IdxVector increments_to_entries(auto const& math_model_param_incrmt) {
+        // construct affected entries
+        std::set<Idx> affected_entries;
```

Comment:
`std::set` is really slow. I think we can perfectly use a `std::vector` here. Just `push_back` all the values. In the end we do one time sort. And another loop copy the unique entries into a new vector.

Reply:
Processed in the new commit, conversation resolved